sopel/modules/ip.py

```diff
@@ -112,80 +302,265 @@ def _find_geoip_db(bot):
+    if any(x in search_str for x in decide):
         # It's an IP/hostname!
-        query = trigger.group(2).strip()
+        query = search_str.strip()
     else:
-        # Need to get the host for the username
-        username = trigger.group(2).strip()
-        user_in_botdb = bot.users.get(username)
-        if user_in_botdb is not None:
-            query = user_in_botdb.host
-
-            # Sanity check - sometimes user information isn't populated yet
-            if query is None:
-                return bot.say("I don't know that user's host.")
+        # Need to get the ip for the username
+        nick = search_str.strip().lower()
+        if user_in_botdb := bot.users.get(nick):
+            if hasattr(user_in_botdb, "ip") and user_in_botdb.ip:
+                query = user_in_botdb.ip
+                # Sanity check - sometimes user information isn't populated yet
+            else:
+                return bot.say("I don't know that user's IP.")
         else:
+            #TODO TODO TODO get from DB
             return bot.say("I\'m not aware of this user.")
 
-    db_path = _find_geoip_db(bot)
-    if db_path is False:
-        LOGGER.error('Can\'t find (or download) usable GeoIP database.')
-        bot.say('Sorry, I don\'t have a GeoIP database to use for this lookup.')
-        return False
+    ex = get_exemption(query).lower()
 
-    if ':' in query:
-        try:
-            socket.inet_pton(socket.AF_INET6, query)
-        except (OSError, socket.error):  # Python 2/3 compatibility
-            return bot.say("[IP/Host Lookup] Unable to resolve IP/Hostname")
-    elif '.' in query:
-        try:
-            socket.inet_pton(socket.AF_INET, query)
-        except (socket.error, socket.herror):
+    if ex:
+        irccloud = "irccloud" in ex
+        mibbit = "mibbit" in ex
+    if not any((irccloud, mibbit)):
+        db_path = _find_geoip_db(bot)
+        if db_path is False:
+            LOGGER.error('Can\'t find (or download) usable GeoIP database.')
+            bot.say('Sorry, I don\'t have a GeoIP database to use for this lookup.')
+            return False
+
+        if ':' in query:
             try:
-                query = socket.getaddrinfo(query, None)[0][4][0]
-            except socket.gaierror:
+                socket.inet_pton(socket.AF_INET6, query)
+            except (OSError, socket.error):  # Python 2/3 compatibility
                 return bot.say("[IP/Host Lookup] Unable to resolve IP/Hostname")
+        elif '.' in query:
+            try:
+                socket.inet_pton(socket.AF_INET, query)
+            except (socket.error, socket.herror):
+                try:
+                    query = socket.getaddrinfo(query, None)[0][4][0]
+                except socket.gaierror:
+                    return bot.say("[IP/Host Lookup] Unable to resolve IP/Hostname")
+        else:
+            return bot.say("[IP/Host Lookup] Unable to resolve IP/Hostname")
```

Comment:
This if-elif-else block seems like it would make sense as a helper function? `resolve_ip_info()` or such